### DISCUSSION TITLE
v1.9.0 - Lots of Bug Fixes plus Aspect Ratio and External Web

### DISCUSSION THREAD

**Discussion by iwalton3** [jellyfin/jellyfin-desktop#397] [General]
The goal of this release was to clean up a bunch of bugs and issues. You can now set the aspect ratio from the player. I also added a bunch of codec profile settings and force transcode Dolby Vision content. Using a web client provided by your Jellyfin server instead of the built-in one is now supported too.

 - Add aspect ratio controls to player. (#388)
 - Don't reset mute between videos. (#349)
 - Auto-detect and switch SteamOS to TV mode. (#237)
 - Add options to disable Windows integrations. (#351 #342)
 - Allow forcing transcodes for HEVC, Hi10p, 4K, HDR, and/or AV1.
     - Media profile settings now work without a client restart.
 - Allow requesting the server to transcode to HEVC.
 - Improve usability of client API to make integrations with other clients easier.
     - New documentation on integrations is [available here](https://github.com/jellyfin/jellyfin-media-player/blob/master/for-web-developers.md).
 - Allow running JMP without an embedded webclient.
     - This will make it easier for upsteam distributions to package the software.
     - You can opt-in to using a server-provided web client through client settings.
     - This may also be useful for people who modify their web client on the server.
 - JMP now uses the official webclient build from repo.jellyfin.org.
 - Switch to [custom mpv build](https://github.com/iwalton3/mpv-winbuild-cmake/releases/tag/v20230304-git-362256e) which has working TLS 1.3 support.
 - Fix time breakage from skipIntroPlugin. (#387)
 - Add delay to prevent broken update dialog. (#373)
 - Prevent Dolby Vision content playing without server transcoding.
    - This fixes colors in media but requires a GPU on the server to support playback.
    - I have disabled remuxing of HEVC due to a bug in the server for now. You can re-enable it in settings but it will break this Dolby Vision fix.
    - If you want to direct play Dolby Vision, you could try using the experimental gpu-next mpv mode with MPV Shim.

If you have issues with media playback due to hardware decoding (green/purple or artifacts), you can [disable it using these instructions](https://github.com/jellyfin/jellyfin-media-player/wiki/Disabling-Hardware-Decoding).

About Windows Downloads:
 - The desktop installer requires administrator rights.
 - You can avoid warnings on files by unblocking the file in the properties dialog.
 - You can manually build a release [using these instructions](https://github.com/jellyfin/jellyfin-media-player#building-for-windows).

<hr /><em>This discussion was created from the release <a href='https://github.com/jellyfin/jellyfin-media-player/releases/tag/v1.9.0'>v1.9.0 - Lots of Bug Fixes plus Aspect Ratio and External Web</a>.</em>

**Comment by senses3** [comment]
>Auto-detect and switch SteamOS to TV mode. (https://github.com/jellyfin/jellyfin-media-player/issues/237)

So I use steam deck mainly in desktop mode and after this update thought something was wrong as the player didnt seem to respond to mouse input the same way it was before. I checked these release notes and realized it went into tv mode by default per the bugfix. I feel dumb not knowing there was a tv mode until just now! :D 

Anyway, maybe look into having JF detect whether it's in gaming mode or desktop mode to choose which to default to? Not quite sure how that would be done.

**Reply by iwalton3** [reply]
It's kind of hard to do with the flatpak sandbox but may be possible. I made tv mode the default because changing the client mode to tv mode through the settings is a major pain in gaming mode, so I wanted to make a more sane default.

**Reply by senses3** [reply]
Since I got my steam deck the one thing i've really had to learn is how flatpaks work and trying to finagle them into functioning as close to a native app as possible. On one hand it's cool being able to run everything in userspace but on the other it's been a bit of a headache all around, especially if you don't have much disk space to work with. 

It really is a great little piece of hardware though.

**Reply by iwalton3** [reply]
For sure I have been very happy with the Steam Deck. Tip you can replace the SSD inside it with a 2230 1TB SSD. I think they even make 2TB ones.

**Reply by senses3** [reply]
I went with a (used from ebay) 512gb kioxia KBG4 as it's way cheaper than 1/2tb nvme currently and in my research it's closest to the stock ssd power consumption I could find.

**Comment by abpgithb** [comment]
"Using a web client provided by your Jellyfin server instead of the built-in one is now supported too."

How does the ForceExternalWebClient work to invoke the Jellyfin Web Client? I clicked it on and restarted both the Media Player and the Server. Playing anything still just seems to play inside the Media Player - am I missing something? I was expecting that it would launch Jellyfin Web on either Chrome or Edge and then play from there.

**Reply by iwalton3** [reply]
No what that does is it causes Jellyfin Media Player to use the web client provided by the server. It still runs inside Jellyfin Media Player as usual. The use case is if you modded your server's web client or if you use JMP packaged by a maintainer that can't bundle the web client.

**Reply by abpgithb** [reply]
Ah, ok. Bummer. I was hoping to be able to leverage NVidia's new VSR update to upscale within the Jellyfin Media Player. (https://blogs.nvidia.com/blog/2023/02/28/rtx-video-super-resolution/)

I know I could login to the server via Chrome or Edge and then play, but the browser doesn't pass through the various HD sound formats.

**Comment by xuhaorss** [comment]
经常下载失败，不知道为什么……是不是有什么特殊的下载方式

**Comment by rems25000** [comment]
Hello, does the Git version "1.9.0" correspond to version 10.8.9-1 of the official repository?

**Reply by iwalton3** [reply]
They are independently versioned, but yes those are the right versions to use together.

**Comment by qvattr0** [comment]
I'm curious: why does JMP not support Dolby Vision (or HDR, for that matter) playback without casting to the MPV Shim?

**Reply by iwalton3** [reply]
Dolby Vision support requires the MPV vo called gpu-next, which uses libplacebo. Unfortunately, since JMP renders media to a surface in a QT application, it cannot use that and must use the mpv output mode which doesn't support Dolby Vision.

MPV Shim can use gpu-next since it doesn't mess with the MPV rendering pipeline. It's also why MPV Shim is more performant and lacks some video output issues that plague JMP.

**Reply by qvattr0** [reply]
I appreciate the swift and detailed response! This seems to be a fairly deep-rooted issue. Would the only way of fixing it or enabling DV/HDR playback on JMP would be to rewrite the software?

**Reply by iwalton3** [reply]
The most likely outcome would be MPV makes an upstream change that allows it. Otherwise it will continue to require server transcoding

**Reply by qvattr0** [reply]
I understand. Well, thank you for this conversation! Hope you have a good day/evening.
